src/flavors/list.rs

```diff
@@ -192,10 +205,13 @@ impl<T> Channel<T> {
                         (*slot).ready.store(true, Ordering::Release);
                     }
                     break;
+                } else {
+                    backoff.step();
                 }
+            } else {
+                // we can help the writer("offset + 1 == BLOCK_CAP") to add next block instead of spinning.
+                self.link_next_block(tail_ptr, tail, tail_index, &guard);
```

Comment:
I think it'd make sense to try linking a new block here only when `offset == BLOCK_CAP`. What do you think?

Reply:
Yes, here only for offset == BLOCK_CAP. Also here mean offset >= BLOCK_CAP, but our algorithm caused only offset == BLOCK_CAP.